client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js

```diff
@@ -74,33 +74,49 @@ export const getForm = ( state, orderId, siteId = getSelectedSiteId( state ) ) =
 	return shippingLabel && shippingLabel.form;
 };
 
-export const getRatesTotal = createSelector(
-	( state, orderId, siteId = getSelectedSiteId( state ) ) => {
-		const form = getForm( state, orderId, siteId );
-		if ( ! form ) {
-			return 0;
-		}
-
-		const { values: selectedRates, available: availableRates } = form.rates;
-
-		const ratesCost = map( selectedRates, ( rateId, boxId ) => {
-			const packageRates = get( availableRates, [ boxId, 'rates' ], false );
+/**
+ * Returns a breakdown of the total price for selected labels in form of { prices, discount, fee, total }
+ * @param {Object} state global state tree
+ * @param {Number} orderId order Id
+ * @param {Number} siteId site Id
+ *
+ * @returns {Object} price breakdown
+ */
+export const getTotalPriceBreakdown = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	const form = getForm( state, orderId, siteId );
+	if ( ! form ) {
+		return null;
+	}
 
-			if ( packageRates ) {
-				const foundRate = find( packageRates, [ 'service_id', rateId ] );
+	const { values: selectedRates, available: availableRates } = form.rates;
+	const prices = [];
+	let discount = 0;
+	let total = 0;
+	for ( const packageId in selectedRates ) {
+		const packageRates = get( availableRates, [ packageId, 'rates' ], false );
+		//if retail_rate is not set on any rate, then the selector should return null
+		if ( ! packageRates || ! packageRates.length || ! isFinite( packageRates[ 0 ].retail_rate ) ) {
```

Comment:
If you're interested, you should be able to pass the `0` and `retail_rates` as part of your call to `get()` above.

Like `get( availableRates, [ packageId, 'rates', 0, 'retail_rate' ], false )` (untested)